s3/main.tf

```diff
@@ -55,76 +71,84 @@ resource "aws_s3_bucket_notification" "log_bucket_notification" {
   }
   depends_on = [aws_s3_bucket_policy.log_bucket]
 }
-
+# This module is to be deprecated and caused many terraform warnings.
+# Attributes have been removed and added as resources.
+# The grants are now passed as a variable to the bucket policy template.
 resource "aws_s3_bucket" "bucket" {
   count         = var.apply_resource == true ? 1 : 0
   bucket        = local.bucket_name
-  acl           = length(var.canonical_user_grants) == 0 ? var.acl : null
   force_destroy = var.force_destroy
+  tags = merge(
+    var.common_tags,
+    tomap(
+      { "Name" = local.bucket_name }
+    )
+  )
+}
 
-  dynamic "grant" {
-    for_each = var.canonical_user_grants
-    content {
-      permissions = grant.value.permissions
-      type        = "CanonicalUser"
-      id          = grant.value.id
-    }
-  }
+resource "aws_s3_bucket_versioning" "bucket_versioning" {
+  count = var.apply_resource == true && length(var.canonical_user_grants) == 0 ? 1 : 0
 
-  versioning {
-    enabled = var.versioning
-  }
+  bucket = aws_s3_bucket.bucket[0].id
 
-  dynamic "lifecycle_rule" {
-    for_each = var.abort_incomplete_uploads == true ? ["include_block"] : []
-    content {
-      id                                     = "abort-incomplete-uploads"
-      enabled                                = true
-      abort_incomplete_multipart_upload_days = 7
-      expiration {
-        days                         = 0
-        expired_object_delete_marker = false
-      }
-    }
+  versioning_configuration {
+    status = "Enabled" # Use "Suspended" to disable versioning
```

Comment:
No need for comment here. This is covered in the Terraform documentation for `versioning_configuration`